test/modules/EdgeHubRestartTester/src/Settings.cs

```diff
@@ -51,9 +46,19 @@ class Settings
             this.TestStartDelay = testStartDelay;
             this.TrackingId = Preconditions.CheckNonWhiteSpace(trackingId, nameof(trackingId));
 
-            if (!(this.DirectMethodEnabled || this.MessageEnabled))
+            this.GetConnectorConfigAsync().Wait();
```

Comment:
Try not to use Wait() method.  Please change ConnectorConfig property as private field; and make GetConnectorConfigAsync as public method; you can cache the result in ConnectorCOnfig private field inside GetConnectorConfigAsync method.